### PR TITLE
Update ID rules in NewVerifiableCredentialBuilder

### PR DESCRIPTION
**Pull request by m-rit**
Address Issue #400 .
 Changes default id in NewVerifiableCredentialBuilder as empty, validates if id is a parsable uri in SetID

### CODE REVIEW

File: credential/builder.go [hunk 1]
```diff
@@ -31,14 +32,15 @@ type VerifiableCredentialBuilder struct {
...
+// Default id is empty
 func NewVerifiableCredentialBuilder() VerifiableCredentialBuilder {
 	contexts := []string{VerifiableCredentialsLinkedDataContext}
 	types := []string{VerifiableCredentialType}
 	return VerifiableCredentialBuilder{
 		contexts: contexts,
 		types:    types,
 		VerifiableCredential: &VerifiableCredential{
-			ID:           uuid.NewString(),
+			ID:           "",
```
Comment (decentralgabe) [inline credential/builder.go:43]: 🤔 given that this is a required property I'm not sure it's a good idea to leave this empty

perhaps it should be added as an optional property to the builder, and if that is empty we generate it automatically?

Comment (m-rit) [reply credential/builder.go:43]: yeah, it's better to keep an option argument. I'll make the changes, thanks!

Comment (m-rit) [reply credential/builder.go]: Hi @decentralgabe, I have added an optional argument to keep ID empty. Kindly review.

File: credential/builder.go [hunk 2]
```diff
@@ -33,19 +33,25 @@ type VerifiableCredentialBuilder struct {
 
 // NewVerifiableCredentialBuilder returns an initialized credential builder with some default fields populated
 // Default id is empty
-func NewVerifiableCredentialBuilder() VerifiableCredentialBuilder {
+func NewVerifiableCredentialBuilder(emptyID ...bool) VerifiableCredentialBuilder {
```
Comment (decentralgabe) [inline credential/builder.go:36]: what do you think about making this a bit more readabile?

we could change the build to accept a new type like `IDvalue` for example


```go

type IDValue string

const EmptyIDValue IDValue = ""

func NewVerifiableCredentialBuilder(id IDValue) VerifiableCredentialBuilder {
    if id == EmptyIDValue {
        // generate ID
    }
    ...
}
```

Comment (m-rit) [reply credential/builder.go:36]: Hi @decentralgabe, thanks for your inputs. 
Please confirm if the behaviour is as follows: 
```
NewVerifiableCredentialBuilder (EmptyIDValue)  -> creates vc with id as uuid
NewVerifiableCredentialBuilder  ("xxx") -> generates vc with id as "xxx"
vcb.SetID("") ->  sets ID as empty
```

Comment (decentralgabe) [reply credential/builder.go:36]: ah, good consideration.

Perhaps we should add....

`EmptyIDValue` -> set ID value to empty
`GenerateIDValue` -> generate a uuid
`CustomIDValue("<your-id-here>")` -> use custom ID

to be unambiguous?

Comment (m-rit) [reply credential/builder.go]: Hi @decentralgabe , I have added IDValue argument here to all three cases (empty ID/ generate ID/ custom ID)

**Review by decentralgabe — APPROVED** [review]

**Comment by decentralgabe** [comment]
thanks @m-rit I approved - you may have to take a look at the build issues

**Comment by m-rit** [comment]
Hi @decentralgabe,  I have fixed all the build issues (the function signature for `NewVerifiableCredentialBuilder` had to be updated in all files). I think additional approval is needed for the workflows.

**Comment by decentralgabe** [comment]
thanks @m-rit approved and merging

**Pull request merged**